File: Ruby/Miscellaneous/isPalindrome/isPalindrome_ans.rb
# This solution takes less time and memory than rebuilding the string backward and
# comparing the two

def isPalindrome?(string)
  length = string.length - 1
   0.upto(string.length - 1) do |i|
     if string[i] != string[length - i]
       return false
     end
  end
  return true
end

#boring answer
def isPalindrome?(string)
  self == self.reverse
end

File: JavaScript/Miscellaneous/isPalindrome/isPalindrome_ans.js
function isPalindrome (string) {
  var length = string.length;

  for (var i = 0; i < length/2; i++) {
    if (string[i] !== string[length - 1 - i]) {
      return false;
    }
  }

  return true;
}


// Time complexity: O(n).
// Space complexity: O(1).
